src/landpatterns/QFP.stanza

```diff
@@ -3,55 +3,190 @@ defpackage jsl/landpatterns/QFP:
   import core
   import jitx
 
+  import jsl/ensure
   import jsl/design/settings
   import jsl/landpatterns/leads
   import jsl/landpatterns/quad
+  import jsl/landpatterns/packages
+  import jsl/landpatterns/pad-planner
+  import jsl/landpatterns/numbering
 
+doc: \<DOC>
+QFN Lead Type
+<DOC>
+public defstruct QFP-Lead <: SMT-Lead:
+  lead-type:LeadProtrusion with: (
+      as-method => true,
+      default => SmallGullWingLeads(), ; TODO: is this OK?
+    )
+  length:Toleranced with: (
+    as-method => true,
+    ensure => ensure-positive!
+    )
+  width:Toleranced with: (
+    as-method => true,
+    ensure => ensure-positive!
+    )
+with:
+  equalable => true
+  hashable => true
+  printer => true
+  constructor => #QFN-Lead
 
-public defn make-qfp-landpattern (
-  num-pins:Int,
-  pitch:Double,
-  ; Lead Parameters
-  lead-length: Toleranced,
-  lead-width: Toleranced,
-  ; Pkg Parameters
-  package-size: Toleranced,
+doc: \<DOC>
+Constructor for `QFP-Lead` type
+<DOC>
+public defn QFP-Lead (
   --
-  density-level: DensityLevel = DENSITY-LEVEL,
-  corner-pads?:False|[Pad, Pad] = false,
-  thermal-pad?: Shape|False = false
+  lead-type:LeadProtrusion,
+  length:Toleranced,
+  width:Toleranced
   ):
-  val lead-type = BigGullWingLeads when pitch >= 0.0625 else SmallGullWingLeads
-  make-quad-landpattern(
-    num-pins, pitch,
-    lead-length, lead-width,
-    package-size, ; lead-span
-    lead-type,
-    package-size,
-    density-level = density-level,
-    corner-pads? = corner-pads?
-    thermal-pad? = thermal-pad?
-  )
+  #QFN-Lead(lead-type, length, width)
 
-public pcb-landpattern qfp-landpattern (
-  num-pins:Int,
-  pitch:Double,
-  ; Lead Parameters
-  lead-length: Toleranced,
-  lead-width: Toleranced,
-  ; Pkg Parameters
-  package-size: Toleranced,
+doc: \<DOC>
+Helper for Creating QFP Lead Profiles
+<DOC>
+public defn QFP-Lead-Profile (
   --
-  density-level: DensityLevel = DENSITY-LEVEL,     ; the density level of the design
-  corner-pads?:False|[Pad, Pad] = false, ; corner pads
-  thermal-pad?: Shape|False = false
+  lead-type : LeadProtrusion
+  span : Toleranced,
+  pitch : Double,
+  lead-length : Toleranced,
+  lead-width : Toleranced,
+  ) -> Quad-Lead-Profile:
+  val lead = QFP-Lead(lead-type = lead-type, length = lead-length, width = lead-width)
+  val profile = Lead-Profile(span = span, pitch = pitch, lead = lead)
+  #Quad-Lead-Profile(profile, None(), profile, None())
+
+public defstruct QFP <: Quad-Package :
+  doc: \<DOC>
+  Number of leads ignoring any depopulated leads and thermal leads
+  <DOC>
+  num-leads:Int with: (ensure => ensure-even-positive!, as-method => true)
+
+  doc: \<DOC>
+  Lead Profiles for all of the QFP's edges
+
+  By default the QFN uses the {@link SmallGullWingLeads} protrusion
```

Comment:
`QFN` -> `QFP`

I think you want this to talk about using the `QFP-Lead` type as the base profile.